[cutlass backend][BE][clean-up] refactor to remove use of autotune_fallback_to_aten=True in cutlass backend tests

henrylhtsang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #152850

Differential Revision: [D74192001](https://our.internmc.facebook.com/intern/diff/D74192001/)

Motivation: clean up post https://github.com/pytorch/pytorch/issues/147479. I plan to leave the rest of the clean-up as an first time issue.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

henrylhtsang committed:
[cutlass backend][BE] remove use of autotune_fallback_to_aten in cutlass backend tests and benchmark



Differential Revision: [D74192001](https://our.internmc.facebook.com/intern/diff/D74192001/)

[ghstack-poisoned]

chenyang78 commented on test/inductor/test_cutlass_backend.py:
```diff
@@ -107,21 +107,19 @@ def mm(a, b):
                 "autotune_in_subproc": True,
-                "max_autotune_gemm_backends": "CUTLASS",
+                # put triton here to make sure we have some choices
+                "max_autotune_gemm_backends": "TRITON,CUTLASS",
```
Wondering why we are making this change? Thanks.

henrylhtsang replied:
So this is actually a good question.

First off, this PR is part of https://github.com/pytorch/pytorch/issues/147479, which wants to deprecate the use of autotune_fallback_to_aten. 

Before this PR, the list of choices is just one aten mm (added since we set autotune_fallback_to_aten=True). It is important for the list to be not empty, otherwise we will hit https://github.com/pytorch/pytorch/blob/main/torch/_inductor/select_algorithm.py#L1866-L1869
(NOTE TO SELF: actually I should see if I can simplify this logic a bit more. Maybe because we are mocking it, it won't raise an assertion. Or maybe I can just catch the error.)

Anyway hence I was adding triton to the list of backend to make sure the list of choices isn't 0. I can't

chenyang78 commented on benchmarks/inductor_backends/cutlass.py:
```diff
@@ -80,7 +80,6 @@ def benchmark_torch_function_in_microseconds(func: Callable, *args, **kwargs) ->
 
 @dataclass(frozen=True, kw_only=True)
 class ExperimentConfig:
-    autotune_fallback_to_aten: bool = False
```
ditto - why are we skipping this path?

henrylhtsang replied:
> ditto - why are we skipping this path?

same as above: This PR is part of https://github.com/pytorch/pytorch/issues/147479, which wants to deprecate the use of autotune_fallback_to_aten. Now autotune_fallback_to_aten defaults to True, so we can remove this.

I plan to leave the rest as a first time issue on github

henrylhtsang committed:
Update on "[cutlass backend][BE][clean-up] refactor to remove use of autotune_fallback_to_aten=True in cutlass backend tests"


Differential Revision: [D74192001](https://our.internmc.facebook.com/intern/diff/D74192001/)

Motivation: clean up post https://github.com/pytorch/pytorch/issues/147479. I plan to leave the rest of the clean-up as an first time issue.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
@pytorchbot merge
